contracts/Chain.sol

```diff
@@ -126,23 +45,19 @@ contract Chain is BaseChain {
         uint8[] memory _v,
         bytes32[] memory _r,
         bytes32[] memory _s
-    ) public { // it could be external, but for external we got stack too deep
-        uint32 lastBlockId = getLatestBlockId();
-        uint32 lastDataTimestamp = squashedRoots[lastBlockId].extractTimestamp();
+    ) external {
+        // below two checks are only for pretty errors, so we can safe gas and allow for raw revert
+        // if (_keys.length != _values.length) revert ArraysDataDoNotMatch();
+        // if (_v.length != _r.length || _r.length != _s.length) revert ArraysDataDoNotMatch();
 
-        unchecked {
-            // we will not overflow with timestamp and padding(uint16)
-            if (lastDataTimestamp + padding >= block.timestamp) revert BlockSubmittedToFast();
-        }
+        _verifySubmitTimestampAndIncSequence(_dataTimestamp);
 
-        if (_dataTimestamp <= lastDataTimestamp) revert DataToOld();
         // we can't expect minter will have exactly the same timestamp
         // but for sure we can demand not to be off by a lot, that's why +3sec
         // temporary remove this condition, because recently on ropsten we see cases when minter/node
         // can be even 100sec behind
         // require(_dataTimestamp <= block.timestamp + 3,
```

Comment:
I wonder if we should  remove all the commented code from contracts

Reply:
we agree to remove in next release